src/db.rs

```diff
@@ -236,8 +236,13 @@ impl DbInner {
         let (tx, rx) = tokio::sync::oneshot::channel();
         let batch_msg = WriteBatchMsg::WriteBatch(WriteBatchRequest { batch, done: tx });
 
-        self.maybe_apply_backpressure().await;
-        self.write_notifier.send(batch_msg).ok();
+        if self.wal_enabled() {
+            self.maybe_apply_wal_backpressure().await;
+        }
+        self.maybe_apply_memtable_backpressure().await;
```

Comment:
instead of gating writes on either the imm wal list or the imm memtable list being full, we could:

1. move this line behind an `else` - that way we apply backpressure at the level that the writer is writing into (wal vs memtable).
2. pause wal flushes when the flush thread sees that there are too many immutable memtables

The benefit is that while l0/memtable is full, we can still write data into the wal until that queue fills up.